src/createAction.ts

```diff
@@ -44,6 +47,17 @@ export interface ActionMixin<T, O extends DoOptions<T>> {
 	 * Disable the task if enabled
 	 */
 	disable(): void;
+
+	/**
+	 * A method which may be called to configure the action after it's been created.
+	 *
+	 * It's up to the implementation to decide what happens if this method is called multiple times. Implementations
+	 * may throw or return a rejected promise.
+	 *
+	 * @param configuration The configuration. Implementations will need to cast to their expected configuration object
+	 * @return May return a promise in case configuration is asynchronous
+	 */
+	configure(configuration: Object): Promise<void> | void;
```

Comment:
Ah 👍  on the index.

I don't think generics add all that much value. [Here](https://github.com/novemberborn/app-example/blob/b71c1472fa88458acb7dce766886ca701f1b0139/src/actions/canCloseTab.ts#L11) the challenging bit is extending `Action` itself to store configuration.

Though if we add an explicit index then perhaps it'll be harder to cast to another type in the `configure()` definition? That would suck.